Backends/CLX/medium.lisp

```diff
@@ -412,9 +428,10 @@ translated, so they begin at different position than [0,0])."))
            (loop for y fixnum from 0 below height do
                 (let ((elt (aref idata y x)))
                   (setf (aref pdata y x) (ash elt -8)))))
-      (unless (or (>= width 2048) (>= height 2048)) ;### CLX bug
-        (xlib:put-image pm pm-gc pm-image :src-x 0 :src-y 0 :x 0 :y 0
-                        :width width :height height))
+      (if (or (>= width 2048) (>= height 2048)) ;### CLX bug
```

Comment:
Since this occurs twice: maybe call `put-image-recursively` unconditionally in both cases? The function call overhead shouldn't matter in this context.